src/pictureNamingTask.ts

```diff
@@ -0,0 +1,236 @@
+    <input type="button" value="Incorrect" onclick="document.getElementById('result').value='Incorrect';">
+    <br>
+    <label for="result">The response was:</label>
+    <input type="text" id="result" name="result" readonly>
+    <hr>
+    <input type="text" id="textBox" name="notes" placeholder="Log any other notes here">
+    <p>Log response and press submit to continue</p>
+  `,
+      button_label: "Submit",
+      autofocus: "textBox",
+      data: {
+        difficultyLevel: jsPsych.evaluateTimelineVariable("difficultyLevel"),
+        correctResponse: jsPsych.evaluateTimelineVariable("correctResponse"),
+      },
+    };
+    const testProcedure = {
+      timeline: [preload, blankPage, showImg, blankPage, logging],
+      timeline_variables: experiment_stimuli,
+      // to reload the experiment_stimuli after one repetition has been completed
+      on_timeline_start: function () {
+        this.timeline_variables = experiment_stimuli;
+      },
+    };
+    timeline.push(testProcedure);
+
+    const loop_node = {
+      timeline: timeline,
+      loop_function: function (data: any) {
+        data = data;
+        // tracking number of corret answers
+        // need to access logging trial info
+        let clearSet = false;
+        if (numberOfTrialsRun < totalNumberOfTrialsToRun) {
+          // getting the most recent logged result
+          const loggingResponseArray: [] = jsPsych.data
+            .get()
+            .filter({ trial_type: "survey-html-form" })["trials"];
+          const lastTrialIndex = loggingResponseArray.length - 1;
+          const lastTrialResults =
+            loggingResponseArray[lastTrialIndex]["response"];
+
+          if (lastTrialResults["result"] === "Correct") {
+            numberOfCorrectAnswers++;
+            clearSet = false;
+          } else if (lastTrialResults["result"] === "Incorrect") {
+            numberOfCorrectAnswers = 0;
+          }
+          // difficulty level logic, <x> correct answers in a row, increase, <y> incorret answer decrease
+          if (numberOfCorrectAnswers === advancementSchedule) {
+            currentDifficultyLevel++;
+            // need to reset as difficulty has changed
+            numberOfCorrectAnswers = 0;
+            clearSet = true;
+          } else if (numberOfCorrectAnswers === regressionSchedule) {
+            if (currentDifficultyLevel > 1) {
+              currentDifficultyLevel--;
+            }
+          }
+          experiment_stimuli = createStimuli(
+            currentDifficultyLevel,
+            language,
+            clearSet,
+          );
+          numberOfTrialsRun++;
+          console.log(`numberOfTrialsRun: ${numberOfTrialsRun}`);
+          return true;
+        } else if (numberOfTrialsRun === totalNumberOfTrialsToRun) {
+          console.log("trial complete");
+          return false;
+        }
+      },
+    };
+    // @ts-ignore
```

Comment:
add comment to justify why this is necessary

Reply:
not necessary. removed